ctapipe/core/qualityquery.py

```diff
@@ -132,11 +144,18 @@ def get_table_mask(self, table):
         mask : np.ndarray[bool]
             Boolean mask of valid entries.
         """
-        n_criteria = len(self.quality_criteria) + 1
+        n_criteria = len(self.criteria_names) + 1
         result = np.ones((n_criteria, len(table)), dtype=bool)
 
-        for i, res in enumerate(self.engine(table), start=1):
-            result[i] = res
+        index_table = Table({"tel_id": table["tel_id"], "index": np.arange(len(table))})
```

Comment:
MMh, not sure we should always assume `tel_id` here... 

Couldn't we add also here the `key` argument? So e.g. `get_table_mask(table, key="tel_id")` would use the telescope-wise value but `get_table_mask(table, key=None)` would use the global default (which might be the only value specified

Reply:
Added the `key` here (which is the column name of the thing to group-by) and use it in the sklearn reconstructors